tests/pytests/integration/ssh/test_log.py

```diff
@@ -0,0 +1,146 @@
+"""
+Integration tests for salt-ssh py_versions
+"""
+import logging
+import socket
+import time
+
+import pytest
+from saltfactories.utils import random_string
+
+from tests.support.helpers import Keys
+
+pytest.importorskip("docker")
+
+
+log = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.skip_if_binaries_missing("dockerd"),
+]
+
+
+def check_container_started(timeout_at, container):
+    sleeptime = 2
+    while time.time() <= timeout_at:
+        time.sleep(sleeptime)
+        try:
+            if not container.is_running():
+                return False
+            sock = socket.socket()
+            sock.connect(("localhost", container.get_check_ports()[22]))
+            return True
+        except Exception as err:  # pylint: disable=broad-except
+            break
+        finally:
+            sock.close()
+        time.sleep(sleeptime)
+        sleeptime *= 2
+    else:
+        return False
+    return False
```

Comment:
```suggestion
def check_container_started(timeout_at, container):
    sleeptime = 2
    while time.time() <= timeout_at:
        time.sleep(sleeptime)
        try:
            if not container.is_running():
                return False
            sock = socket.socket()
            sock.connect(("localhost", container.get_check_ports()[22]))
            return True
        except Exception as err:  # pylint: disable=broad-except
            break
        finally:
            sock.close()
        time.sleep(sleeptime)
        sleeptime *= 2
    else:
        return False
    return False
```
Salt factories checks are not enough?

https://github.com/saltstack/pytest-salt-factories/blob/master/src/saltfactories/daemons/container.py#L656-L691

Reply:
I can probably remove this. Was a copy/paste. will remove

Reply:
If we have this logic somewhere else, we should remove that too.

Reply:
Not necessarily on this PR

Reply:
It honestly might actually just be solved by the extra time.sleep and is just good timing. Didn't realize that function was getting called. Let me dive in a bit more.